### PR TITLE
Add Regional Team and STP pages

### PR DESCRIPTION
**Pull request by inglesp**

### CODE REVIEW

**Comment by HelenCEBM** [comment]
STP described as regional team:
image
http://staging.openprescribing.net/stp/E54000019/

**Comment by sebbacon** [comment]
Inconsistent capitalisation - can we pick one?

image

image

**Comment by HelenCEBM** [comment]
STP and regional team breakdown by CCGs talks about practices, and for STPs could have "STP" appended to the title?
http://staging.openprescribing.net/stp/E54000043/doacs/
http://staging.openprescribing.net/regional-team/Y55/arb/

image

**Comment by HelenCEBM** [comment]
Exclude non-CCGs from list of orgs in Regional teams? 
http://staging.openprescribing.net/regional-team/Y55/

image

**Comment by HelenCEBM** [comment]
`Looking for more data` refers to analyse page but we don't have STP/regional team analyses enabled there yet - assume that will be a future development.
http://staging.openprescribing.net/regional-team/Y55/
http://staging.openprescribing.net/stp/E54000040/

image

Suggest either change to `for your CCGs and practices`, truncate to `...on prescribing data.`, remove reference altogether, OR if easy, pre-fill analyse form with all the CCGs in STP/regional team (assume tricky!)

**Comment by HelenCEBM** [comment]
Supplementary to previous comment, we could add under `Looking for more data?` something like
`These Regional team / STP pages are a new feature. We have more data for individual CCGs/practices [links] and for the whole of England [link] (such as our price-per-unit tool, GBGs etc)`

**Comment by brianmackenna** [comment]
This is beautiful and going to be immense. STPs will love us!

**Menu Layout**

See [user feedback](https://docs.google.com/document/d/1h2rtZi3Snx5DquL0ooG5F0tIlyIbkwAF88tfPBixcpc/edit?usp=sharing) on menu layout. I think we go

- Area and Practice Dashboards (include all England here)
- Tools (including analyse unless we keep seperate until how to is populated)
- How to? (not sure as we only have one so far, suggest leave out until we have 3or4)

**STP Layout**

- GBG, PPU, Price concessions etc. Do we need the matrix store of just for a later date?
- When you break down to CCG level (see [here](http://staging.openprescribing.net/stp/E54000027/diltiazem/)) it does not allow you to "Split the charts into individual GP". This would be useful

**Regional Layout**

- Deciles are weird here but we can launch and improve later. I'll need to talk this through with people so I can articulate in public correctly.
- When breaking down by region it jumps next to CCG. I'm wondering could you split to constituent STPs first?
- Comments on GP breakdown as above (see London example [here](http://staging.openprescribing.net/regional-team/Y56/test_strip/l))

**Comment by inglesp** [comment]
@sebbacon @evansd Suggest you hold off reviewing until I've addressed feedback, and fixed tests.

**Comment by inglesp** [comment]
I've addressed all feedback from @sebbacon and @HelenCEBM, and this is now all on staging.

**Comment by inglesp** [comment]
As for @brianmackenna's feedback:

* I've updated "Dashboards" to "Area and Practice Dashboards", but I haven't done the rest of the work in #1396 (Menu reorganisation).
* GBG, PPU, Price concessions etc -- I've not begun to look at how we implement these for STPs/Regional teams, but have created #1540, #1541, #1542, #1543 to track.
* "Split the charts into individual GP" -- do you mean that you'd like to be able to see a measure for all practices in a STP/Regional Team?  Given the number of practices in some STPs/Regional Teams I'm not sure how practical this is.  Could we return to this later if there's demand from users?
* "Deciles are weird here" -- yeah but we'll have to wait for the MatrixStore to land before we can do anything less weird.
* "When breaking down by region it jumps next to CCG. I'm wondering could you split to constituent STPs first?"  ODS are now publishing data linking STPs to Regional Teams, so once we're importing data from there this would be possible.  See #1539.

**Comment by inglesp** [comment]
@sebbacon, @evansd, this is now ready to be reviewed.

**Comment by HelenCEBM** [comment]
sorry i spotted another thing! `ccg` should be replaced with `Regional Team`
image

**Comment by brianmackenna** [comment]
> "Split the charts into individual GP" -- do you mean that you'd like to be able to see a measure for all practices in a STP/Regional Team? Given the number of practices in some STPs/Regional Teams I'm not sure how practical this is. Could we return to this later if there's demand from users?

I meant to break it down to a practice for an individual CCG. On the yellow highlight it would be good to have _Split the measure into charts for individual practices in this CCG_ As you say I don't think there would be demand across an STP to breakdown


image

**Comment by HelenCEBM** [comment]
- On the individual presentation breakdown pages the drop-down menus don't seem to work, e.g. http://staging.openprescribing.net/measure/icsdose/regional-team/Y54/

- The links to CCG breakdowns for individual measures for STPs or regional teams (as highlighted below) go to an error 500 (though I'm sure these worked before) http://staging.openprescribing.net/stp/E54000034/libre/
image
image

**Comment by inglesp** [comment]
@HelenCEBM, thanks for looking at this.

The menu dropdowns haven't been broken by this PR (see #1544) although we should fix them before this PR is merged, because more stuff is now only accessible via a menu.

And I've fixed the pages giving 500 server error.  (I'd been careless with pulling in other changes that had been made to the code since I began working on this.)

File: openprescribing/media/js/src/measure_utils.js [hunk 1]
```diff
@@ -97,8 +97,17 @@ var utils = {
   _rollUpByOrg: function(data, orgType) {
     var rolled = {};
     _.each(data.data, function(d) {
-      var id = (orgType === 'practice') ? d.practice_id : d.pct_id;
-      var name = (orgType === 'practice') ? d.practice_name : d.pct_name;
+      var id;
+      var name;
+
+      if (orgType == 'ccg') {
```
[the remark this answers is not in the document]
Comment (inglesp) [reply openprescribing/media/js/src/measure_utils.js]: Do we ever run the linter?

**Comment by inglesp** [comment]
Thanks for the review!

Have addressed review comments in https://github.com/ebmdatalab/openprescribing/compare/0f6be010e749cf91aa4662d55058490e161952cd..c5e31f6f55debaf076fc3cf59886cbc71249b80b.

Have also squashed all the commits, as the commit history didn't make much sense.

We may want to co-ordinate launching this with a blog, but as far as I'm concerned this is good to go, so you could merge it while I'm away next week.

**Comment by brianmackenna** [comment]
On the [region dashboard](https://openprescribing.net/regional-team/Y63/) it lists 9 NHS England regions.

NHS England is currently reorganising and will have more regions probably which is probably reflected in ONS ODS codes that @inglesp  has used. However we only have 5 regions with CCGs attached to them. Could 4 without data be suppressed for now?
https://openprescribing.net/regional-team/Y61/
https://openprescribing.net/regional-team/Y60/
https://openprescribing.net/regional-team/Y63/
https://openprescribing.net/regional-team/Y62/

**Pull request merged**